app/src/androidTest/java/com/github/se/project/ui/components/SubjectSelectorTest.kt

```diff
@@ -16,35 +16,20 @@ class SubjectSelectorTest {
 
   @get:Rule val composeTestRule = createComposeRule()
 
-  @Test
-  fun selectingSubjectUpdatesCorrectly() {
-    val selectedSubject = mutableStateOf<Subject>(Subject.NONE)
-    composeTestRule.setContent { SubjectSelector(selectedSubject = selectedSubject) }
-
-    // Initially, dropdown items are not displayed
-    Subject.entries.forEach { subject ->
-      if (subject == Subject.NONE) return@forEach
-      composeTestRule.onNodeWithTag("dropdown${subject.name}").assertDoesNotExist()
-    }
-
-    // Click on the subject button to open the dropdown
-    composeTestRule.onNodeWithTag("subjectButton").performClick()
-
-    // **Wait for the UI to settle after the click**
-    composeTestRule.waitForIdle()
+  private val selectedSubjects = mutableStateListOf<Subject>()
 
-    // Now, dropdown items should be displayed
-    Subject.entries.forEach { subject ->
-      if (subject == Subject.NONE) return@forEach
-      composeTestRule.onNodeWithTag("dropdown${subject.name}").assertIsDisplayed()
+  @Before
+  fun setUp() {
+    composeTestRule.setContent {
+      SubjectSelector(selectedSubjects = selectedSubjects, multipleSelection = true)
```

Comment:
This code is indeed shared by all test so it is great to write it only once in the setup method! 
Maybe we could also add a waitForIdle() to be sure we do not test before the screen is ready!